### PR TITLE
[MI-2417] Add test cases for client.go

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

File: server/plugin/client_test.go [hunk 1]
```diff
@@ -561,3 +562,181 @@ func TestUpdateStateOfRecordInServiceNowClient(t *testing.T) {
...
+		description        string
+		statusCode         int
+		expectedStatusCode int
+		errorMessage       error
+		expectedErr        string
+	}{
+		{
+			description:        "GetMe: user don't exist on ServiceNow",
+			statusCode:         http.StatusNotFound,
+			expectedStatusCode: http.StatusNotFound,
```
Comment (raghavaggarwal2308) [inline server/plugin/client_test.go:583]: Can't we use a single variable here?

File: server/plugin/client_test.go [hunk 2]
```diff
@@ -561,3 +562,181 @@ func TestUpdateStateOfRecordInServiceNowClient(t *testing.T) {
...
+		description        string
+		statusCode         int
+		expectedStatusCode int
+		errorMessage       error
+		expectedErr        string
+	}{
+		{
+			description:        "CreateIncident: valid",
+			statusCode:         http.StatusOK,
+			expectedStatusCode: http.StatusOK,
```
Comment (raghavaggarwal2308) [inline server/plugin/client_test.go:624]: same

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/client_test.go [hunk 3]
```diff
@@ -561,3 +520,168 @@ func TestUpdateStateOfRecordInServiceNowClient(t *testing.T) {
...
+		ServiceNowBaseURL: "mockServiceNowBaseURL",
+	})
+	for _, testCase := range []struct {
+		description  string
+		statusCode   int
+		errorMessage error
+		expectedErr  string
+	}{
+		{
+			description: "GetMe: user don't exist on ServiceNow",
```
Comment (manojmalik20) [inline server/plugin/client_test.go:538]:
```suggestion
			description: "GetMe: user doesn't exist on ServiceNow",
```

File: server/plugin/client_test.go [hunk 4]
```diff
@@ -561,3 +520,168 @@ func TestUpdateStateOfRecordInServiceNowClient(t *testing.T) {
...
+			statusCode:  http.StatusNotFound,
+			expectedErr: "user doesn't exist on ServiceNow instance mockServiceNowBaseURL with email mockEmail",
+		},
+		{
+			description:  "GetMe: with error",
```
Comment (manojmalik20) [inline server/plugin/client_test.go:543]: Update the description to something meaningful.

File: server/plugin/client_test.go [hunk 5]
```diff
@@ -561,3 +520,168 @@ func TestUpdateStateOfRecordInServiceNowClient(t *testing.T) {
...
+		statusCode   int
+		errorMessage error
+		expectedErr  string
+	}{
+		{
+			description: "CreateIncident: valid",
+			statusCode:  http.StatusOK,
+		},
+		{
+			description:  "CreateIncident: with error",
```
Comment (manojmalik20) [inline server/plugin/client_test.go:580]: Same here.

File: server/plugin/client_test.go [hunk 6]
```diff
@@ -561,3 +520,168 @@ func TestUpdateStateOfRecordInServiceNowClient(t *testing.T) {
...
+		expectedErr  string
+	}{
+		{
+			description: "GetIncidentFieldsFromServiceNow: valid",
+			statusCode:  http.StatusOK,
+		},
+		{
+			description:  "GetIncidentFieldsFromServiceNow: with latest update set not uploaded",
+			statusCode:   http.StatusBadRequest,
+			errorMessage: errors.New("mockError: Requested URI does not represent any resource"),
```
Comment (manojmalik20) [inline server/plugin/client_test.go:662]: Wasn't there a constant present for this error message?

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**